test/framework/docker_logcollector.go

```diff
@@ -31,8 +32,15 @@ import (
 // DockerLogCollector collect logs from a CAPD workload cluster.
 type DockerLogCollector struct{}
 
+func machineContainerName(cluster, machine string) string {
+	if strings.HasPrefix(machine, cluster) {
+		return machine
+	}
+	return fmt.Sprintf("%s-%s", cluster, machine)
+}
```

Comment:
Can we add a comment on why is this needed? And in the PR title as well please

Reply:
done!